cloudsync/tasks_test.py

```diff
@@ -229,3 +234,59 @@ def test_stream_to_s3_no_video():
     with pytest.raises(Video.DoesNotExist):
         stream_to_s3(999999)  # pylint: disable=no-value-for-parameter
+
+
+@pytest.mark.django_db
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_monitor_watch(mocker, user):  # pylint: disable=unused-argument,redefined-outer-name
+    """Test the Watch bucket monitor task"""
+    user = UserFactory(username='admin')  # pylint: disable=unused-variable
+    mocker.patch.multiple('cloudsync.tasks.settings',
+                          ET_PRESET_IDS=('1351620000001-000061', '1351620000001-000040', '1351620000001-000020'),
+                          AWS_REGION='us-east-1', ET_PIPELINE_ID='foo')
+    mock_encoder = mocker.patch('cloudsync.api.VideoTranscoder.encode')
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    assert s3c.get_object(Bucket=bucket.name, Key=upload) is not None
+    monitor_watch_bucket.delay()
+    mock_encoder.assert_called_once()
+    new_video = Video.objects.get(title=upload)
+    new_videofile = new_video.videofile_set.get(encoding=EncodingNames.ORIGINAL)
+    assert new_videofile.bucket_name == settings.VIDEO_S3_BUCKET
+    with pytest.raises(ClientError):
+        s3c.get_object(Bucket=bucket.name, Key=upload)
+
+
+@pytest.mark.django_db
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_monitor_watch_badname(mocker):  # pylint: disable=unused-argument,redefined-outer-name
+    """
+    Test no video is created for a file with a bad name, but other filenames are processed
+    """
+    user = UserFactory(username='admin')  # pylint: disable=unused-variable
+    mocker.patch.multiple('cloudsync.tasks.settings',
+                          ET_PRESET_IDS=('1351620000001-000061', '1351620000001-000040', '1351620000001-000020'),
+                          AWS_REGION='us-east-1', ET_PIPELINE_ID='foo')
+    mock_encoder = mocker.patch('cloudsync.api.VideoTranscoder.encode')
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_BUCKET)
+    uploads = ('MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4',
+               'Bad Name.mp4',
+               'MIT-6.046-lec-mit-0000-2017apr06-0404.mp4')
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    for filename in uploads:
+        bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), filename)
+    monitor_watch_bucket.delay()
+    assert mock_encoder.call_count == 2
+    assert not Video.objects.filter(title=uploads[1])
+    assert Video.objects.get(title=uploads[0])
+    assert Video.objects.get(title=uploads[2])
```

Comment:
ok, I understand, but, if you can, it is better to avoid re-testing the entire logic of the backend because it is really difficult to understand the difference.